presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -379,27 +379,38 @@ public TableStatistics getTableStatistics(ConnectorTransactionHandle transaction
         tableStatistics.setRowCount(calculateRowsCount(partitionStatisticsMap));
 
         for (Map.Entry<String, ColumnHandle> columnEntry : tableColumns.entrySet()) {
+            String columnName = columnEntry.getKey();
+            HiveColumnHandle hiveColumnHandle = (HiveColumnHandle) columnEntry.getValue();
             com.facebook.presto.spi.statistics.ColumnStatistics.Builder columnStatistics = com.facebook.presto.spi.statistics.ColumnStatistics.builder();
-            columnStatistics.setDistinctValuesCount(calculateDistinctValuesCount(partitionStatisticsMap, columnEntry.getKey()));
-            columnStatistics.setNullsCount(calculateNullsCount(partitionStatisticsMap, columnEntry.getKey()));
-            tableStatistics.setColumnStatistics(columnEntry.getValue(), columnStatistics.build());
+            if (hiveColumnHandle.isPartitionKey()) {
```

Comment:
I'm wondering if we could have a strategy for computing null and unique value counts (two separate methods) with two implemenatations (for columns that are partition keys and for other columns)?

Alternarively, a strategy with a single method computing a `ColumnSummary {nullsCount, uniqueValuesCount }`? (one method; two strategy implementations)